subm/src/request1bld.js

```diff
@@ -0,0 +1,220 @@
+/* eslint-disable no-await-in-loop */
+// See https://github.com/Agoric/validator-profiles/wiki/Request-1-BLD
+
+const { DiscordAPI, getContent, paged } = require('./discordGuild.js');
+const { upsert } = require('./sheetAccess');
+const { searchBySender, transfers } = require('./tendermintRPC');
+
+const config = {
+  host: 'rpc-agoric.nodes.guru',
+  address: 'agoric15qxmfufeyj4zm9zwnsczp72elxsjsvd0vm4q8h',
+  quorum: 2,
+};
+
+const fail = () => {
+  throw Error();
+};
+
+/**
+ * @param {ReturnType<ReturnType<typeof DiscordAPI>['channels']>} channel
+ * @param {ReturnType<ReturnType<typeof DiscordAPI>['guilds']>} guild
+ * @param {Snowflake} role
+ * @param {number} quorum
+ * @yields {{ message: Message, endorsers: User[] }}
+ * @typedef {import('./discordGuild').Snowflake} Snowflake
+ */
+async function* authorizedRequests(channel, guild, role, quorum) {
+  /** @type {Map<Snowflake, import('./discordGuild').GuildMember>} */
+  const memberDetail = new Map();
+  /** @param {Snowflake} id */
+  const getMemberDetail = async id => {
+    if (memberDetail.has(id)) {
+      return memberDetail.get(id) || fail();
+    }
+    const detail = await guild.members(id);
+    // console.log(detail);
+    memberDetail.set(id, detail);
+    return detail;
+  };
+
+  const messages = await paged(channel.getMessages);
+  const hasAddr = messages.filter(msg => msg.content.match(/agoric1/));
```

Comment:
need tighter regex match

Reply:
need? really? I think no regex is needed at all; this is just an optimization to avoid processing most messages. The 2 reviewers are the real gating factor.

Reply:
true as the extract is lower - i figured it would be possible to copy the same regex up here to tighten down the search